examples/http_mitm_proxy_boring.rs

```diff
@@ -231,9 +228,25 @@ async fn http_mitm_proxy(ctx: Context, req: Request) -> Result<Response, Infalli
 
     // NOTE: use a custom connector (layers) in case you wish to add custom features,
     // such as upstream proxies or other configurations
-    let mut client = EasyHttpWebClient::default()
-        .with_http_conn_req_inspector(UserAgentEmulateHttpConnectModifier::default())
-        .with_http_conn_req_inspector((
+
+    let base_tls_config = if let Some(hello) = ctx
+        .get::<SecureTransport>()
+        .and_then(|st| st.client_hello())
+        .cloned()
+    {
+        // TODO once we fully support building this from client hello directly remove this unwrap
+        TlsConnectorDataBuilder::try_from(hello).unwrap()
+    } else {
+        TlsConnectorDataBuilder::new_http_auto()
+    };
+    let base_tls_config = base_tls_config.with_server_verify_mode(ServerVerifyMode::Disable);
+
+    let client = EasyHttpWebClient::builder()
+        .with_tls_proxy_support_using_boringssl()
```

Comment:
how come this doens't accept a config but the non-proxy one does? I get that you want to have a "default" one using the "default" config, but that goes for the non-proxy version as well, no? next to an alt method that does take a config?